catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorTest.java

```diff
@@ -135,6 +155,35 @@ public void testMoveFile() throws Exception {
     }
 
     @Test
+    public void testBlackListedFileExtension() throws Exception {
+        System.setProperty("bad.file.extensions", ".txt");
+        monitor = createContentDirectoryMonitor();
+        submitConfigOptions(monitor, monitoredDirectoryPath, ContentDirectoryMonitor.MOVE);
+        doAndVerifyFileDidNotMove(monitoredDirectory, monitoredDirectory, "input1.txt");
+        System.setProperty("bad.file.extensions", "");
+    }
+
+
+    @Test
+    public void testBlackListedFile() throws Exception {
+        System.setProperty("bad.files", "input1.txt");
+        monitor = createContentDirectoryMonitor();
+        submitConfigOptions(monitor, monitoredDirectoryPath, ContentDirectoryMonitor.MOVE);
+        doAndVerifyFileDidNotMove(monitoredDirectory, monitoredDirectory, "input1.txt");
+        System.setProperty("bad.files", "");
+    }
+
+    @Test
+    public void testBlackListedFileExtensions() throws Exception {
+        System.setProperty("bad.file.extensions", ".txt,.bin");
+        monitor = createContentDirectoryMonitor();
+        submitConfigOptions(monitor, monitoredDirectoryPath, ContentDirectoryMonitor.MOVE);
+        doAndVerifyFileDidNotMove(monitoredDirectory, monitoredDirectory, "input1.txt");
+        doAndVerifyFileDidNotMove(monitoredDirectory, monitoredDirectory, "input1.bin");
+        System.setProperty("bad.file.extensions", "");
+    }
```

Comment:
❗️ Missing a test where the properties are present but empty, and will result in state variables being null.